lock/src/seqlock.rs

```diff
@@ -170,6 +177,25 @@ impl<'s, T> ReadGuard<'s, T> {
         mem::forget(self);
         result
     }
+
+    pub fn upgrade(&self) -> Option<WriteGuard<'s, T>> {
```

Comment:
I'm not sure about its API...
 - Called by shared reference: `ReadGuard` is still safe to exist, but it will always fail to upgrade. Is it better to consume the guard?
 - Returning `Option`. Is it better to return `Result`?

Reply:
- I prefer moving `self`.
- I prefer `Result`.